### PR TITLE
Kennethbranch

### PR DESCRIPTION
**Pull request by kennllanto**

### CODE REVIEW

File: Snake/BusinessLayer/Engine.cs [hunk 1]
```diff
@@ -23,6 +23,7 @@ public class Engine : IDisposable
         AI newAI;
         bool smartMove = false;
         AI newSmartAI;
+        Laser AILaser;
```
Comment (mattrumble) [inline Snake/BusinessLayer/Engine.cs]: Should the lazer be part of the AI?

ie AI.Lazer.Shoot() etc.....

File: Snake/BusinessLayer/AI.cs [hunk 2]
```diff
@@ -9,6 +9,14 @@ namespace BusinessLayer
     public class AI
     {
         private Direction currentDirection;
+        public Direction AIDirection
```
Comment (danaterr) [inline Snake/BusinessLayer/AI.cs]: consider next time not using a backing field and writing the property like this:
public Direction AIDirection{ get; private set;}

File: Snake/BusinessLayer/Laser.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    public class Laser : Point
+    {
+        Direction currentDirection;
+
+        public Laser(int newX, int newY, Direction direction) : base (newX, newY)
+        {
+            setX(newX);
+            setY(newY);
+            currentDirection = direction;
+
```
Comment (danaterr) [inline Snake/BusinessLayer/Laser.cs]: whitespacing

File: Snake/SnakeTest/Laser_Test.cs [hunk 4]
```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace SnakeTest
+{
+    [TestClass]
+    public class Laser_Test
```
Comment (danaterr) [inline Snake/SnakeTest/Laser_Test.cs]: don't use underscores for class name even in tests

File: Snake/SnakeTest/Laser_Test.cs [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    {
+        [TestMethod]
+        public void Move_Test_if_Laser_Moves_Down()
```
Comment (danaterr) [inline Snake/SnakeTest/Laser_Test.cs]: no need for Test in the name of method

File: Snake/SnakeTest/Laser_Test.cs [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        {
+
+            BusinessLayer.Laser testLaser = new BusinessLayer.Laser(2,2, BusinessLayer.Direction.Down);
+
+            testLaser.Move();
+
+            Assert.IsTrue(testLaser.returnX() == 3);
```
Comment (danaterr) [inline Snake/SnakeTest/Laser_Test.cs]: this seems odd. the coordinates are 2,2 but the test looks for 3. something is not right

File: Snake/SnakeTest/Laser_Test.cs (hunk 4 above)
Comment (kennllanto) [inline Snake/SnakeTest/Laser_Test.cs]: Done

File: Snake/SnakeTest/Laser_Test.cs (hunk 5 above)
Comment (kennllanto) [inline Snake/SnakeTest/Laser_Test.cs]: modified

**Pull request merged**